### PR TITLE
Task List functionality

### PR DESCRIPTION
**Pull request by TWLTriston**
Addresses #60, based on Poll code. Needs peer review, has been tested briefly in MP

### CODE REVIEW

File: tasklist.lua [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+		x = x + 1
+	end
+end
+
+
+local function tasklist_refresh()
+
+	local x = 1
+
```
Comment (Valansch) [inline tasklist.lua]: What does this x do?

File: tasklist.lua [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+	global.tasklist_author = player.name
+
+	local msg = player.name
+	msg = msg .. " has created an updated tasklist!"
+
+	local frame = player.gui.left["tasklist-assembler"]
+	frame.destroy()
+
+	local x = 1
```
Comment (Valansch) [inline tasklist.lua]: And this one

File: tasklist.lua [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+		if (frame) then
+				frame.destroy()
+		end
+
+		if (global.autoshow_tasklist_for_player[player.name] == true) then
+			tasklist_show(player)
+		end
+
+		player.print(msg)
```
Comment (Valansch) [inline tasklist.lua]: Dont you want to use game.print here?

File: tasklist.lua (hunk 1 above)
Comment (TWLTriston) [reply tasklist.lua]: Was used in a while loop, updating to a foreach loop

File: tasklist.lua (hunk 3 above)
Comment (TWLTriston) [reply tasklist.lua]: Yes, moving out of the loop

File: tasklist.lua (hunk 2 above)
Comment (TWLTriston) [reply tasklist.lua]: Cleaning up the same while vs for logic

File: tasklist.lua [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+	tasklist_panel_button_table.tasklist_hide_button.style.font = "default-bold"
+	tasklist_panel_button_table.tasklist_hide_button.style.minimal_height = 38
+	-- Possible for later preferences tasklist_panel_button_table.add { type = "checkbox", caption = "Show Tasklist", state = global.autoshow_tasklist_for_player[player.name], name = "auto_show_tasklist_checkbox"	}
+end
+
+local function tasklist(player)
+	local frame = player.gui.left["tasklist-assembler"]
+	frame = frame.table_tasklist_assembler
+
+	global.tasklist_items = {"","","","",""}
```
Comment (Valansch) [inline tasklist.lua]: You dont need to initalize those :D lua tables dont have a fixed size

Comment (TWLTriston) [reply tasklist.lua]: The table needs to be primed, else later references to draw the table blows up

We can either check for ~nil or prime it.  This method avoids the stop off for a check, and instead knows a value will be there

Comment (Valansch) [reply tasklist.lua]: Oh okay, yes i was assuming the texts are not nil.

**Comment by TWLTriston** [comment]
Should be all set, corrected the nil reference that was being thrown

**Pull request merged**